To: vim_dev@googlegroups.com
Subject: Patch 7.3.226
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.226
Problem:    On a 64 bit system "syn sync fromstart" is very slow.  (Bjorn
            Steinbrink)
Solution:   Store the state when starting to parse from the first line.
Files:      src/syntax.c


*** ../mercurial/vim73/src/syntax.c	2011-05-25 17:56:23.000000000 +0200
--- src/syntax.c	2011-06-19 04:47:08.000000000 +0200
***************
*** 559,565 ****
      if (INVALID_STATE(&current_state))
      {
  	syn_sync(wp, lnum, last_valid);
! 	first_stored = current_lnum + syn_block->b_syn_sync_minlines;
      }
      else
  	first_stored = current_lnum;
--- 559,571 ----
      if (INVALID_STATE(&current_state))
      {
  	syn_sync(wp, lnum, last_valid);
! 	if (current_lnum == 1)
! 	    /* First line is always valid, no matter "minlines". */
! 	    first_stored = 1;
! 	else
! 	    /* Need to parse "minlines" lines before state can be considered
! 	     * valid to store. */
! 	    first_stored = current_lnum + syn_block->b_syn_sync_minlines;
      }
      else
  	first_stored = current_lnum;
*** ../vim-7.3.225/src/version.c	2011-06-19 04:31:54.000000000 +0200
--- src/version.c	2011-06-19 04:49:53.000000000 +0200
***************
*** 711,712 ****
--- 711,714 ----
  {   /* Add new patch number below this line */
+ /**/
+     226,
  /**/

-- 
Press any key to continue, press any other key to quit.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
